src/libraries/System.Drawing.Common/src/System/Drawing/DrawingComWrappers.cs

```diff
@@ -0,0 +1,283 @@
+                return inst.LockRegion(libOffset, cb, dwLockType);
+            }
+
+            [UnmanagedCallersOnly]
+            private static Interop.HRESULT UnlockRegionImplementation(IntPtr thisPtr, ulong libOffset, ulong cb, uint dwLockType)
+            {
+                Interop.Ole32.IStreamComWrapper inst = ComInterfaceDispatch.GetInstance<Interop.Ole32.IStreamComWrapper>((ComInterfaceDispatch*)thisPtr);
+                return inst.UnlockRegion(libOffset, cb, dwLockType);
+            }
+
+            [UnmanagedCallersOnly]
+            private static Interop.HRESULT StatImplementation(IntPtr thisPtr, out Interop.Ole32.STATSTG pstatstg, Interop.Ole32.STATFLAG grfStatFlag)
+            {
+                Interop.Ole32.IStreamComWrapper inst = ComInterfaceDispatch.GetInstance<Interop.Ole32.IStreamComWrapper>((ComInterfaceDispatch*)thisPtr);
+                inst.Stat(out pstatstg, grfStatFlag);
+                return Interop.HRESULT.S_OK;
+            }
+
+            [UnmanagedCallersOnly]
+            private static Interop.HRESULT CloneImplementation(IntPtr thisPtr, IntPtr* ppstm)
+            {
+                if (ppstm == null)
+                {
+                    return Interop.HRESULT.STG_E_INVALIDPOINTER;
+                }
+
+                Interop.Ole32.IStreamComWrapper inst = ComInterfaceDispatch.GetInstance<Interop.Ole32.IStreamComWrapper>((ComInterfaceDispatch*)thisPtr);
+
+                *ppstm = Instance.GetOrCreateComInterfaceForObject(inst.Clone(), CreateComInterfaceFlags.None);
+                return Interop.HRESULT.S_OK;
+            }
+        }
+
+#pragma warning disable CS0649 // fields are never assigned to
+        internal struct IPictureVtbl
+        {
+            public IUnknownVtbl IUnknownImpl;
+            public IntPtr GetHandle;
+            public IntPtr GetHPal;
+            public IntPtr GetPictureType;
+            public IntPtr GetWidth;
+            public IntPtr GetHeight;
+            public IntPtr Render;
+            public IntPtr SetHPal;
+            public IntPtr GetCurDC;
+            public IntPtr SelectPicture;
+            public IntPtr GetKeepOriginalFormat;
+            public IntPtr SetKeepOriginalFormat;
+            public IntPtr PictureChanged;
+            public delegate* unmanaged<IntPtr, IntPtr, int, int*, int> SaveAsFile;
+            public IntPtr GetAttributes;
+            public IntPtr SetHdc;
+        }
+
+        internal struct VtblPtr
+        {
+            public IntPtr Vtbl;
+        }
+#pragma warning restore CS0649
+
+        internal interface IPicture : IDisposable
+        {
+            static readonly Guid IID = new Guid(0x7BF80980, 0xBF32, 0x101A, 0x8B, 0xBB, 0, 0xAA, 0x00, 0x30, 0x0C, 0xAB);
+
+            // NOTE: Only SaveAsFile is invoked. The other methods on IPicture are not necessary
+
+            int SaveAsFile(IntPtr pstm, int fSaveMemCopy, int* pcbSize);
```

Comment:
At this point this `IPicture` interface could be considered the C# friendly version. That means you could change the signature to be nicer and hide all the interop goo in the implementation as opposed to it leaking out of the wrapper.

```csharp
internal interface IPicture : IDisposable
{
    static readonly Guid IID = new Guid(0x7BF80980, 0xBF32, 0x101A, 0x8B, 0xBB, 0, 0xAA, 0x00, 0x30, 0x0C, 0xAB);

    void SaveAsFile(GPStream stream, bool fSaveMemCopy, out int pcbSize);
}
```

Reply:
I just realized I suggested changing this to an `int*` &ndash; sorry about that.

Reply:
Since there is only a single caller, and the "interop goo" isn't so bad here, I left the `IPicture` as a 1-1 blittable projection of the native interface.